wadm/lib/wadm/observer/observer.ex

```diff
@@ -0,0 +1,37 @@
+defmodule Wadm.Observer do
+  @doc """
+  This module receives event publications from a lattice observer from the
+  lattice_observer library. That library does no pre-filtering, so it's up to
+  us to determine if the downstream consumers of this information need to know.
+
+  It's important to note that this module isn't stateful nor is it a genserver.
+  There will be 1 monitor process per lattice prefix, so the more of those that exist,
+  the more lattice traffic will pass through this 'state_changed' function.
+  """
+  use LatticeObserver.Observer
+  import Wadm.Deployments.DeploymentMonitor, only: [is_reconciliation_ready?: 1]
+  require Logger
+
+  @impl true
+  def state_changed(old_state, new_state, event, lattice_prefix) do
```

Comment:
Generally speaking if we are making this declarative, we shouldn't pass old state. All that matters is how the new state differs from the current state. Based on my experience from having reconciliation loops in k8s, passing old_state into a handler leads to bad coding practices where people try to carry all sorts of weird state in the struct/event data that should just be available from the system

Reply:
yeah the function name `state_changed` is misleading. It is actually invoked every time an event is received regardless of whether it made a difference in state. The reason I passed both was to allow the watchers/consumers to decide if the state change made in the processing of the event was worth acting upon or not.

Reply:
The more I re-read your comment the more I think I should change the latticeobserver behaviour so that the callback is called `event_received` (or something similar) and it only contains the new/current state, event, and prefix and not the old state.